neqo-http3/src/connection_client.rs

```diff
@@ -151,21 +151,16 @@ impl Http3Client {
         &self.conn.odcid().expect("Client always has odcid")
     }
 
-    /// Returns a resumption token if present.
     /// A resumption token encodes transport and settings parameter as well.
-    #[must_use]
-    pub fn resumption_token(&mut self) -> Option<Vec<u8>> {
-        if let Some(token) = self.conn.resumption_token() {
-            if let Some(settings) = self.base_handler.get_settings() {
-                let mut enc = Encoder::default();
-                settings.encode_frame_contents(&mut enc);
-                enc.encode(&token[..]);
-                Some(enc.into())
-            } else {
-                None
-            }
-        } else {
-            None
+    fn create_resumption_token(&mut self, token: ResumptionToken) {
+        if let Some(settings) = self.base_handler.get_settings() {
+            let mut enc = Encoder::default();
+            settings.encode_frame_contents(&mut enc);
+            enc.encode(&token.token[..]);
```

Comment:
Can we add a method on ResumptionToken to append/prepend some additional information?  We're doing this in a couple of places now and it would be good to have common code for creating a new token from an existing one with some extra stuff added in.